internal/uvm/vsmb.go

```diff
@@ -402,11 +404,11 @@ func (vsmb *VSMBShare) Clone(ctx context.Context, vm *UtilityVM, cd *cloneData)
 		guestPath:       vsmb.guestPath,
 		serialVersionID: vsmbCurrentSerialVersionID,
 	}
-
+	shareKey := getVSMBShareKey(vsmb.HostPath, vsmb.options.ReadOnly)
```

Comment:
Did VSMB not work before this fix?

Reply:
No it did. But it didn't store the VSMB share in the vsmb map with the correct key. I accidentally found this while looking for some other bug.

Reply:
Was there any negative impact without this fix? Trying to understand if this is something we should properly document in its own commit.

Reply:
As far as I understand there should be no negative impact even without this fix. The `getVSMBShareKey` is useful only when you are trying to mount the same VSMB share once as read-only and once as read/write. But cloning is only used to clone the uvm image VSMB share (i.e the "os" share) and the template container image share. And none of these shares will again be mounted as read/write VSMB shares. 
So this fix is just to make sure that all VSMB shares in the vsmb share map follow the same pattern of keys.